File: src/Client/Logistics.OfficeApp/src/app/core/api/models/truck/truck.dto.ts
import {AddressDto} from "../address.dto";
import {EmployeeDto} from "../employee/employee.model";
import {GeoPointDto} from "../geo-point.dto";
import {LoadDto} from "../load/load.dto";
import {TruckStatus, TruckType} from "./enums";

export interface TruckDto {
  id: string;
  number: string;
  type: TruckType;
  status: TruckStatus;
  mainDriver?: EmployeeDto;
  secondaryDriver?: EmployeeDto;
  currentAddress?: AddressDto;
  currentLocation?: GeoPointDto;
  loads?: LoadDto[];
}

File: src/Shared/Logistics.Shared.Models/Truck/TruckDto.cs
using Logistics.Domain.Primitives.Enums;
using Logistics.Domain.Primitives.ValueObjects;

namespace Logistics.Shared.Models;

public record TruckDto
{
    public Guid? Id { get; set; }
    public string? Number { get; set; }
    public int VehicleCapacity { get; set; }
    public TruckType Type { get; set; }
    public TruckStatus Status { get; set; }
    public Address? CurrentAddress { get; set; }
    public GeoPoint? CurrentLocation { get; set; }
    public EmployeeDto? MainDriver { get; set; }
    public EmployeeDto? SecondaryDriver { get; set; }
    public IEnumerable<LoadDto> Loads { get; set; } = [];
}
